### DISCUSSION TITLE
Questions about the API `/core/groups/`

### DISCUSSION THREAD

**Discussion by xunleii** [goauthentik/authentik#6652] [Q&A] [answered]
Hello,

I have a question regarding the `/core/groups/` path on Authentik. I quickly generated a [Python client for Authentik](https://github.com/radiofrance/authentik-client-python) in order to use it for our internal tools written in Python.
Unfortunately, it is unusable on `GET /core/groups/` because, when a group has no parent, the mandatory `parent_name` field is not sent. After reading the official API browser, I can confirm that `parent` can be `null`, but that `parent_name` must be set.

Is this a bug?

_NOTE: about the python client, we haven't used it yet, so we can't validate that it works perfectly. However, we can transfer ownership of the repository if and when you wish (it was created with this in mind)._

**Comment by BeryJu** [comment, marked as answer]
This particular quirk has been around for a while, it basically comes down to API fields that are declared as `read_only=True` that are not required, as they are still marked as required in the schema.

Would this diff fix the generation?

```diff
diff --git a/schema.yml b/schema.yml
index daf3a9eba..26edc61bd 100644
--- a/schema.yml
+++ b/schema.yml
@@ -30498,6 +30498,7 @@ components:
         parent_name:
           type: string
           readOnly: true
+          nullable: true
         users:
           type: array
           items:
```

**Reply by xunleii** [reply]
Thanks a lot for your response, making the `parent_name` field `nullable` fix this issue (https://github.com/radiofrance/authentik-client-python/commit/413d9c1f27edef1f95cfc9ac99e2213f6142ee92#diff-440e52d4b33e2fe6e8a94f84ec999e6f0b767d47d9b89fc1cc03ad4bbef55753).

This is something that must be done on the Authentik side or on the client library side _(like an issue with the generator or a patch to apply everytime the client is built)_?

**Reply by BeryJu** [reply]
This will be included in 2023.8.2+; see https://github.com/goauthentik/authentik/pull/6700
